modules/user/user-snapshot.service.test.ts

```diff
@@ -0,0 +1,1438 @@
+                                gaugeBalances: [],
+                                farms: [],
+                                farmBalances: [],
+                            },
+                        ],
+                    }),
+                );
+            }),
+        ],
+    );
+
+    const snapshotsFromService = await userService.getUserBalanceSnapshotsForPool(userAddress, poolId1, 'THIRTY_DAYS');
+    //check if 4th snapshot has been inferred from three present ones
+    expect(snapshotsFromService.length).toBe(4);
+    const snapshotsFromDb = await prisma.prismaUserPoolBalanceSnapshot.findMany({
+        where: {
+            userAddress: userAddress,
+        },
+        include: { pool: true },
+    });
+
+    // check if the 3 snapshots have been persisted
+    expect(snapshotsFromDb.length).toBe(3);
+
+    // check if balances are calculated correctly
+    expect(snapshotsFromService[0].timestamp).toBe(threeDaysAgo);
+    expect(snapshotsFromService[0].walletBalance).toBe('1');
+    expect(snapshotsFromService[1].timestamp).toBe(twoDaysAgo);
+    expect(snapshotsFromService[1].walletBalance).toBe('0');
+    expect(snapshotsFromService[1].totalValueUSD).toBe('0');
+    expect(snapshotsFromService[1].fees24h).toBe('0');
+    expect(snapshotsFromService[1].percentShare).toBe(0);
+
+    expect(snapshotsFromService[2].timestamp).toBe(oneDayAgo);
+    expect(snapshotsFromService[2].walletBalance).toBe('1');
+
+    expect(snapshotsFromService[3].timestamp).toBe(today);
+    expect(snapshotsFromService[3].walletBalance).toBe('1');
+
+    const poolSnapshots = await prisma.prismaPoolSnapshot.findMany({
+        where: { poolId: poolId1 },
+    });
+
+    // check if usd value, percent share of the pool and fees are correctly calculated based on poolsnapshots
+    for (const userBalanceSnapshot of snapshotsFromService) {
+        let foundPoolSnapshot = false;
+        for (const poolSnapshot of poolSnapshots) {
+            if (poolSnapshot.timestamp === userBalanceSnapshot.timestamp) {
+                expect(userBalanceSnapshot.totalValueUSD).toBe(
+                    `${poolSnapshot.sharePrice * parseFloat(userBalanceSnapshot.totalBalance)}`,
+                );
+                expect(userBalanceSnapshot.percentShare).toBe(
+                    parseFloat(userBalanceSnapshot.totalBalance) / poolSnapshot.totalSharesNum,
+                );
+                expect(userBalanceSnapshot.fees24h).toBe(
+                    `${
+                        userBalanceSnapshot.percentShare *
+                        poolSnapshot.fees24h *
+                        (1 - networkConfig.balancer.swapProtocolFeePercentage)
+                    }`,
+                );
+                foundPoolSnapshot = true;
+            }
+        }
+        //make sure we have a pool snapshot for each user snapshot
+        expect(foundPoolSnapshot).toBe(true);
+    }
+});
+
+test('user left pool, no 0 snapshots returned', async () => {
```

Comment:
use case would be what you write under behaviour under test: 'when user leaves pool, no more stats are returned' or something